### PR TITLE
Add machine lifecycle diagram

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR adds machine lifecycle diagram to the proposal

### CODE REVIEW

**Comment by enxebre** [comment]
can you also please commit the source code for the diagram so it can be updated in follow ups

**Comment by enxebre** [comment]
do we want to commit the png file? some of the paths should actually go back to the initial step as you do for when "Add finalisers" wdyt?

**Comment by alexander-demichev** [comment]
You're right, they should, but I'm afraid this diagram can become a labyrinth if I do it for every re-queue action

**Comment by JoelSpeed** [comment]
I think the PNG file should probably be committed rather than relying on GitHub userimages

How many requeues are currently being omitted?

**Comment by elmiko** [comment]
agreed about committing the png file

**Comment by alexander-demichev** [comment]
I included PNG in the commit.

@JoelSpeed All requeues that are caused by CRUD action failures, etc. The diagram represents the logic of this function https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/machine/controller.go#L144

File: enhancements/machine-api/machine-instance-lifecycle.md [hunk 1]
```diff
@@ -135,6 +135,10 @@ For it to happen in a non disruptive manner for actuators this proposal is inten
 
 Once this is settled we might consider to simplify the interface.
 
+### Machine lifecycle diagram
+
+![image](https://user-images.githubusercontent.com/20123872/74226366-b9069380-4cbc-11ea-9e51-b1ccc2a1b35b.png)
```
Comment (JoelSpeed) [inline enhancements/machine-api/machine-instance-lifecycle.md]: Can we link this to the checked in image? Do relative image links work? 🤔

Comment (alexander-demichev) [reply enhancements/machine-api/machine-instance-lifecycle.md]: They should, thanks for pointing at it

File: enhancements/machine-api/machine-instance-lifecycle.md [hunk 2]
```diff
@@ -135,6 +135,10 @@ For it to happen in a non disruptive manner for actuators this proposal is inten
 
 Once this is settled we might consider to simplify the interface.
 
+### Machine lifecycle diagram
+
```
Comment (enxebre) [inline enhancements/machine-api/machine-instance-lifecycle.md]: can we please make this a sub-section of the "Implementation Details" After "Errors"?

**Pull request merged**